src/main/groovy/com/vherasymenko/avro/encoder/core/CourseInstallEncoderService.groovy

```diff
@@ -76,9 +86,16 @@ class CourseInstallService implements CourseInstallPort {
         // encode document using avro json encoding
         def encodedDocument = encoder.encodeEvent( schema, avroData, CourseInstall.class )
 
+        // save schema to the schema registry server
+        def registryResponse = registryClient.register( schema.name, 'avro', schema.toString() )
+        log.info( "The schema with subject ${registryResponse.schemaReference.subject} was saved to the schema registry " +
+                "with the id ${registryResponse.id}." )
+        def schemaId = registryResponse.id
```

Comment:
It does give a better context, but I believe the MessageHeaders.SCHEMA_ID is also providing the same context that we see here.